modules/ximgproc/src/sparse_match_interpolators.cpp

```diff
@@ -173,15 +173,55 @@ Ptr<EdgeAwareInterpolatorImpl> EdgeAwareInterpolatorImpl::create()
 void EdgeAwareInterpolatorImpl::interpolate(InputArray from_image, InputArray from_points, InputArray, InputArray to_points, OutputArray dense_flow)
 {
     CV_Assert( !from_image.empty() && (from_image.depth() == CV_8U) && (from_image.channels() == 3 || from_image.channels() == 1) );
-    CV_Assert( !from_points.empty() && from_points.isVector() &&
-               !to_points  .empty() && to_points  .isVector() &&
-               from_points.sameSize(to_points) );
-
+    CV_Assert( !from_points.empty() && !to_points.empty() && from_points.sameSize(to_points) );
+    CV_Assert(from_points.isVector() || from_points.isMat());
+    CV_Assert(to_points.isVector() || to_points.isMat());
     w = from_image.cols();
     h = from_image.rows();
 
-    vector<Point2f> from_vector = *(const vector<Point2f>*)from_points.getObj();
-    vector<Point2f> to_vector   = *(const vector<Point2f>*)to_points  .getObj();
+    vector<Point2f> from_vector;
+    vector<Point2f> to_vector;
+    int npoints = 0;
+    if (from_points.isVector()) {
```

Comment:
This should be enough for provided tests:

```
    vector<Point2f> from_vector;
    vector<Point2f> to_vector;
    from_points.getMat().reshape(2).copyTo(from_vector);
    to_points.getMat().reshape(2).copyTo(to_vector);
```

Reply:
Actually I wanted to avoid to copy the points if not needed. To make the algorithm as fast as possible. Do you know if the copyTo makes a deep or shallow copy

Reply:
`.copyTo()` does deep copy ("deep" doesn't make sense in this statement because `Point2f = { float, float }`).

(code from the patch also performs copy, including unsafe `.getObj()` case)

---

`std::vector` is not able to work in shallow copy mode (without copying data). `std::swap()` is not acceptable here to initialize vector.

if you want a "shallow" copy, you need to use `Mat` / `Mat_<Point2f>` (which can wrap passed data) in this function instead:
```.diff
-vector<Point2f> from_vector;
+Mat_<Point2f> from_mat(from_points.getMat().reshape(2));
```

Reply:
Thanks for that hint. I will implement the later version.